bug: /trace returns 500 — Jinja filter 'format_tok' not registered

## Finding
GET /trace returns HTTP 500. Template trace_tree.html uses `format_tok` and `format_cost` Jinja filters, but no filters are registered on the Jinja2Templates environment anywhere in the codebase (`grep filters\[` returns nothing).

This breaks the P0 'Icicle + indented tree views' requirement — the Tree page is unreachable.

## Evidence
- burnmap/templates/pages/trace_tree.html:90 `{{ trace.total_tokens | int | format_tok }}`
- burnmap/templates/pages/trace_tree.html:92 `{{ trace.total_cost | format_cost }}`
- burnmap/api/web.py:46 — handler renders the template
- Server traceback: `jinja2.exceptions.TemplateAssertionError: No filter named 'format_tok'`
- Live: `curl http://localhost:7820/trace` → 500

## PRD Ref
05-requirements.md — Tree: 'Icicle + indented views over same data' (P0)

--- a/burnmap/api/web.py
+++ b/burnmap/api/web.py
@@ -14,9 +14,23 @@ except ImportError:
 
 _templates_dir = Path(__file__).parent.parent / "templates"
 
+def _format_tok(value: int) -> str:
+    """Format token count: 1_234 → '1.2k', <1000 → plain number."""
+    if value >= 1000:
+        return f"{value / 1000:.1f}k"
+    return str(value)
+
+
+def _format_cost(value: float) -> str:
+    """Format cost in dollars: '$0.0042'."""
+    return f"${value:.4f}"
+
+
 if _FASTAPI:
     router = APIRouter()
     templates = Jinja2Templates(directory=str(_templates_dir))
+    templates.env.filters["format_tok"] = _format_tok
+    templates.env.filters["format_cost"] = _format_cost
 
     def _html(request: Request, template: str, **ctx: object) -> HTMLResponse:
         return templates.TemplateResponse(request, template, ctx)

